org.eclipse.dawnsci.hdf5.model.test/src/org/eclipse/dawnsci/hdf5/model/internal/HierarchicalDataFileModelTest.java

```diff
@@ -306,35 +319,38 @@ public void testReadDatasetFirst() {
 				model.getPath("/entry1/instrument/analyser/cps@units"));
 	}
 
-	private Object createNexusFile(final int dtype, final long[] shape, final Object buffer) throws Exception {
+	private Object createNexusFile(final int dtype, final int[] shape, final Object buffer) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)